### DISCUSSION TITLE
Images published through the API are showing up as white

### DISCUSSION THREAD

**Discussion by albattran** [bluesky-social/atproto#1891] [Bluesky Lexicons]
Hey everyone,
Are there certain maximum specs for uploading images through the API besides the 1MB limit?
When I upload certain PNG files (less than 500K in size) they are showing up as a white canvas on bsky.app
The dimensions for the images are 4989x7639, I uploaded other images with similar dimensions and they work fine
Also if crop the faulty images by a certain amount they work (e.g. 4146x3505 works, but 4889x3524 doesn't work)

Any guidance if we should scale down images would be appreciated.
Thanks so much for your help!

**Comment by albattran** [comment]
Also here is a sample post with the PNG file showing up all white
https://bsky.app/profile/feditst.bsky.social/post/3kf6mbmyub22t

**Reply by DavidBuchanan314** [reply]
It looks like you're using the legacy blob reference format there - which ought to still work, but it is deprecated, so that could be related to the issue.

The post actually refuses to load for me at all, I get a "Error: Failed to parse response body: TypeError: img.image.ref is undefined" in the web UI

**Reply by albattran** [reply]
Thanks David, I didn't realize it was deprecated so soon :)
OK, I tried again with the new API....

Still same issue unfortunately, here is the new post:
https://bsky.app/profile/feditst.bsky.social/post/3kf73yv6jya23

99.99% of our posts work, but the problem appears very specific to these images.

Not sure why it doesn't load for you (you mean bsky.app?) I tried from a different account and it loaded for me.
But your answer gave me an idea to dig further, and this URL: 
https://cdn.bsky.app/img/feed_thumbnail/plain/did:plc:svwrwmdybtcrm3wrgpo4mwxm/bafkreiepxbeebeq7tpykvki54v2skpr5cwyocokghia6d2odr7ylsf34em@jpeg
is giving HTTP 422 "Invalid source image"

**Reply by AshGlidewell** [reply]
I'm having a similar issue here: 
https://bsky.app/profile/vndev.bsky.social/post/3kfqijyt26e2b

I'm following the example that I found in https://atproto.com/blog/create-post#images-embeds

Did you happen to find a solution, @albattran?

**Reply by albattran** [reply]
@AshGlidewell, unfortunately no
The code has been working for months for images, but it just happens with a certain ones (They happen to be black/white, not sure if true for all of them)
Do you also experience the same thing? (mostly working except for certain images, black/white?)

**Reply by AshGlidewell** [reply]
This is unfortunately my first time jumping into it so I haven't tested other images yet, mostly just trying to get the one online before I take a shot at a larger project involving image posting.  The latest one I tried to post was a scaled down version of the below image we also cross-posted to Tumblr.

https://64.media.tumblr.com/c2977e86711a5ec149b6ad250174c6fb/2cca26fda5bf5c97-72/s1280x1920/84b17cab38571908df1c99659bf0c14f4b7a6f66.png

So I guess it's not exactly apples to apples, sorry to pester.

**Reply by devinivy** [reply]
I wonder if you may be running into a case similar to the one answered here: https://github.com/bluesky-social/atproto/discussions/1777

**Reply by AshGlidewell** [reply]
It seems similar in the way that it presents a blank image embed on the post, but I don't think it's being caused by the image dimensions.  The images I've uploaded have been significantly smaller than what's included there.  I'll run another test today with a different image and see if I can reproduce with different image sizes and dimensions.

**Reply by AshGlidewell** [reply]
This is an image I uploaded

```
{
    "blob": {
        "$type": "blob",
        "ref": {
            "$link": "bafkreifobhirsjz5ajnlnxcv3emxwrzqhbga4vuiwthk5ab6wr7v7zyjla"
        },
        "mimeType": "image/png",
        "size": 82487
    }
}
```

And this is what I sent to make this message:
https://bsky.app/profile/vndev.bsky.social/post/3kgbrnajpme2z


```
{
    "collection": "app.bsky.feed.post",
    "repo": "did:plc:7yovtpsbz3jfyzde2pjmkuig",
    "record": {
        "text": "This week's #VNDev discussion! What techniques do you use to make sure your visual novel's dialogue flows smoothly?",
        "createdAt": "2023-12-11T15:50:58.7726684Z",
        "type": "app.bsky.feed.post",
        "embed": {
            "$type": "app.bsky.embed.images",
            "images": [
                {
                    "alt": "This week's #VNDev discussion! What techniques do you use to make sure your visual novel's dialogue flows smoothly?",
                    "image": {
                        "$type": "blob",
                        "ref": {
                            "$link": "bafkreifobhirsjz5ajnlnxcv3emxwrzqhbga4vuiwthk5ab6wr7v7zyjla"
                        },
                        "mimeType": "image/png",
                        "size": 82487
                    }
                }
            ]
        }
    }
}
```

The issue might be something more obvious to someone else, but I'm not seeing it.    Could very well just be in the structure of my json??

**Reply by pixelhustler** [reply]
I'm facing the same issue of posts with images showing up blank, did you ever get a solution?
